packages/webapp/src/components/Form/Unit/useUnit.jsx

```diff
@@ -59,32 +73,52 @@ const useUnit = ({
     setShowError(!!error && !disabled && isDirty);
   }, [error]);
 
-  const { displayUnit, displayValue, options, databaseUnit, isSelectDisabled, measure } =
-    useMemo(() => {
-      const databaseUnit = defaultValueUnit ?? unitType.databaseUnit;
-      const options = getOptions(unitType, system);
-      const hookFormValue = hookFormGetValue(name);
-      const value = hookFormValue || (hookFormValue === 0 ? 0 : defaultValue);
-      const isSelectDisabled = options.length <= 1;
-      const measure = convert().describe(databaseUnit)?.measure;
-      return to && convert().describe(to)?.system === system
-        ? {
-            displayUnit: to,
-            displayValue:
-              defaultValue && roundToTwoDecimal(convert(value).from(databaseUnit).to(to)),
-            options,
-            databaseUnit,
-            isSelectDisabled,
-            measure,
-          }
-        : {
-            ...getDefaultUnit(unitType, value, system, databaseUnit),
-            options,
-            databaseUnit,
-            isSelectDisabled,
-            measure,
-          };
-    }, []);
+  const {
+    displayUnit,
+    displayValue,
+    options,
+    databaseUnit,
+    defaultIsSelectDisabled,
+    reactSelectWidth,
+    dividerWidth,
+    onKeyDown,
+    measure,
+  } = useMemo(() => {
+    const databaseUnit = defaultValueUnit ?? unitType.databaseUnit;
+    const options = getOptions(unitType, system);
+    const hookFormValue = hookFormGetValue(name);
+    const value = hookFormValue || (hookFormValue === 0 ? 0 : defaultValue);
+    const defaultIsSelectDisabled = options.length <= 1;
+    const measure = convert().describe(databaseUnit)?.measure;
+    const reactSelectWidth = getReactSelectWidth(measure);
+    const onKeyDown = getOnKeyDown(measure);
+    const dividerWidth = defaultIsSelectDisabled
+      ? reactSelectWidth - DEFAULT_SELECT_ARROW_ICON_WIDTH
+      : reactSelectWidth;
+
+    return to && convert().describe(to)?.system === system
+      ? {
+          displayUnit: to,
+          displayValue: defaultValue && roundToTwoDecimal(convert(value).from(databaseUnit).to(to)),
+          options,
+          databaseUnit,
+          defaultIsSelectDisabled,
+          measure,
+          reactSelectWidth,
+          dividerWidth,
+          onKeyDown,
+        }
+      : {
+          ...getDefaultUnit(unitType, value, system, databaseUnit),
+          options,
+          databaseUnit,
+          defaultIsSelectDisabled,
+          measure,
+          reactSelectWidth,
+          dividerWidth,
+          onKeyDown,
+        };
+  }, []);
```

Comment:
@SayakaOno 
useMemo should have dependency list right?

If dependency list is big should useMemo be broken up into several useMemo so it is not re-run often?

Reply:
This useMemo returns default values for the Unit component, so it should only be called once. I will add some comments in the next PR to make it clear!

Reply:
I will look into it more for next PR. But only calling it once seems to defeat the purpose of useMemo.

Reply:
I understand what you mean. Please let me know if you come up with a better solution!
Thank you for reviewing :)